File: python/leet121.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if prices is None or len(prices) == 0: return 0;

        max_profit = 0;
        buy_price = prices[0];
        for i in range(1, len(prices)):
            if max_profit < prices[i] - buy_price:
                max_profit = prices[i] - buy_price;
            if prices[i] < buy_price:
                buy_price = prices[i];

        return max_profit;



if __name__ == '__main__':
    sol = Solution();
    profit = sol.maxProfit([7, 1, 5, 3, 6, 4]);
    print(profit);

File: java/leetcode-practice/src/test/java/com/mamta/leetcode/twopointer/Leet121.java
package com.mamta.leetcode.twopointer;

import org.junit.jupiter.api.Test;

public class Leet121 {

    public int maxProfit(int[] prices) {
        if (prices == null || prices.length == 0) return 0;
        int buyPrice = prices[0];
        int profit = 0;
        for(int i=1; i<prices.length;i++) {

            if(profit < prices[i]-buyPrice ) {
                profit = prices[i]-buyPrice;
            }
            if(prices[i] < buyPrice ) {
                buyPrice = prices[i];
            }


        }
        return profit;

    }

    @Test
    public void solution() {
        int[] prices = {7,1,5,3,6,4};

      System.out.println(maxProfit(prices));

    }
}
